lib/vyasa/medium/event.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Vyasa.Medium.Event do
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  alias Vyasa.Written.{Verse, Source}
+  alias Vyasa.Medium.Voice
+
+  @primary_key {:id, Ecto.UUID, autogenerate: true}
+  schema "events" do
+    field :origin, :integer
+    field :duration, :integer
+    field :phase, :string
+
+    embeds_many :fragments, EventFrag do
+      field :offset, :integer
+      field :duration, :integer
+      field :quote, :string
+      field :status, :string
+    end
+
+    belongs_to :verse, Verse, foreign_key: :verse_id, type: :binary_id
+    belongs_to :voice, Voice, foreign_key: :voice_id, type: :binary_id
+    belongs_to :source, Source, foreign_key: :source_id, type: :binary_id
+  end
+
+  @doc false
+  def changeset(event, attrs) do
+    event
+    |> cast(attrs, [:origin, :duration, :phase, :verse_id, :voice_id, :source_id])
```

Comment:
+ minor nit: add "interim" phase or something for the non-start and non-end phases of an event